runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -346,11 +346,16 @@ public V get(@ConflictParameter Object key) {
     <I extends Comparable<I>>
     Collection<Entry<K, V>> getByIndex(@Nonnull IndexName indexName, I indexKey) {
         String secondaryIndex = indexName.get();
-        Map<K, V> res = secondaryIndexes.get(secondaryIndex).get(indexKey);
-
-        return res == null ?
-                Collections.emptySet():
-                new HashSet<>(res.entrySet());
+        try {
+            Map<K, V> res = secondaryIndexes.get(secondaryIndex).get(indexKey);
```

Comment:
Can't you determine this ahead of time (i.e. before the access) instead of catching the exception post access?

Reply:
I'm aware of the PassThrough annotation but I'm not actually sure on how to combine both, to run this check previous to the accessor.

Reply:
Do we have an annotation that supports this behavior, or would I need to modify the annotationProcessor and add a new annotation for this?

Reply:
Hmm, "before the access" as in before the java object access and not SMR access. Do you catch a NPE because secondaryIndexes.get(secondaryIndex) can return null ?

Reply:
I thought you were referring to the SMR access, which also made sense to me, to avoid going all the way through if no actual index exists. I did the change.